### DISCUSSION TITLE
"team_foreign_key" in "model_has_permissions" purpose

### DISCUSSION THREAD

**Discussion by bkaczmarekdev** [spatie/laravel-permission#2335] [Q&A] [answered]
Hi, I'm wondering what's the purpose of team_foreign_key in model_has_permissions. In fact I wanted to use it, so that I can give direct permission to User with particular team_id. Unluckily doing
`$user->givePermissionTo(['name' => 'resource.permission', 'team_id' => 1]);`
doesn't work - returns 

> Integrity constraint violation: 1048 Column 'team_id' cannot be null

Can I make it work, or is there any other purpose of team_foreign_key in this table?

**Comment by erikn69** [comment, marked as answer]
**Documentation:** [working-with-teams-permissions](https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions#content-working-with-teams-permissions)
```diff
-$user->givePermissionTo(['name' => 'resource.permission', 'team_id' => 1]);
+setPermissionsTeamId(1); // Set team_id globally for the package
+$user->givePermissionTo('resource.permission');
```
Also
```php
    'teams' => true,
```
https://github.com/spatie/laravel-permission/blob/490fa00b0c8286659a534d822f7e74c684c86d27/config/permission.php#L114

**Reply by bkaczmarekdev** [reply]
Thanks, sorry for taking your time if it was all about reading documentation. I must've misread something about creating and granting roles. Thank you again.
